challenges/functions.js

```diff
@@ -6,26 +6,50 @@
   * The last parameter accepts a callback 
   * In the body of the function return the callback with the two parameters that you created
 */
+function consume(param1, param2, callback){
+  return callback(param1, param2);
+}
+function consumeCallback(item1, item2){
+  console.log(item1+' consume '+item2);
+}
 
+consume('kids', 'time', consumeCallback);
 
 /* Step 2: Create several functions to callback with consume();
   * Create a function named add that returns the sum of two numbers
   * Create a function named multiply that returns the product of two numbers 
   * Create a function named greeting that accepts a first and last name and returns "Hello first-name last-name, nice to meet you!"
 */
 
+function add(a,b) {
+  console.log(a + b);
+}
+function multiply(a,b) {
+  console.log(a * b);
+}
+function greeting(first, last) {
+  this.first = first;
+  this.last = last;
+  console.log(`Hello ${this.first} ${this.last}, nice to meet you!`);
+}
 
 /* Step 3: Check your work by un-commenting the following calls to consume(): */
-// consume(2,2,add); // 4
-// consume(10,16,multiply); // 160
-// consume("Mary","Poppins", greeting); // Hello Mary Poppins, nice to meet you!
+consume(2,2,add); // 4
+consume(10,16,multiply); // 160
+consume("Mary","Poppins", greeting); // Hello Mary Poppins, nice to meet you!
 
 
 // ==== Closures ==== 
 
 // Explain in your own words why `nestedfunction()` can access the variable `internal`.
 
-// Explanation: 
+// Explanation:   
+
+
+
+// *** nestedFunction has access to `internal` because it is a method of myFunction, it treats myFunction as if it were the global object ***
```

Comment:
Because of closure internal can be used because nestedFunction will go out of scope to get use of that variable